modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java

```diff
@@ -693,7 +693,11 @@ public String getTypeDeclaration(Schema p) {
         if (ModelUtils.isArraySchema(p)) {
             ArraySchema ap = (ArraySchema) p;
             Schema inner = ap.getItems();
-            return getSchemaType(p) + "[" + getTypeDeclaration(inner) + "]";
+            String innerDeclaration = getTypeDeclaration(inner);
+            if (inner.getNullable() != null && inner.getNullable()) {
```

Comment:
suggestion: you may consider using `Boolean.TRUE.equals(inner.getNullable())` instead to avoid the null check and make the code more readable.